src/hal/time.cpp

```diff
@@ -122,27 +122,26 @@ time_t OswHal::getTime(time_t& offset) {
     return this->getUTCTime() + offset;
 }
 
-void OswHal::getDate(time_t& offset, uint32_t* day, uint32_t* weekDay) {
-    RtcDateTime d = RtcDateTime();
-    d.InitWithUnix32Time(this->getTime(offset));
-    *weekDay = d.DayOfWeek();
-    *day = d.Day();
-}
+void OswHal::getDate(time_t& offset, OSW_DATE* oswDate, uint32_t* setWDay) {
+    const char* dayMap[7] = {
+        LANG_SUNDAY
+        , LANG_MONDAY
+        , LANG_TUESDAY
+        , LANG_WEDNESDAY
+        , LANG_THURSDAY
+        , LANG_FRIDAY
+        , LANG_SATURDAY
+    };
```

Comment:
This is bad. You have an array, allocated on the stack and return a pointer to a value on it. This may be optimized by the compiler, but you are potentially returning a pointer, which becomes invalid with returning from the function / leaving the scope. This issue existed beforehand. Either you make the array `static`, so it will be moved into the `BSS`-section of memory, or you have to find another solution. The current solution is very dangerous...

Reply:
Also, you merge the functions `getDate` and `getWeekday`. But...

* the parameter is still called `setWDay`
* the previous function `getWeekday` allowed you to just get the weekday name OR ask for a specific weekday name - now you always have to provide an `OswTime` (also **nullptr check is missing**) and you can only receive the weekday using this struct

Maybe do not merge these functions?

Reply:
Also, why did you add the weekday-name to the struct? It is not being used elsewhere (like in `getUTCTime` or `getTime`) and only used here.

Reply:
The more I think about it... Maybe just remove the whole `getWeekday`-function and potentially replace it with an `const std::array<const char*>`.

Reply:
...you added the array, but are not using it here?